metadata-ingestion/tests/integration/grafana/test_grafana.py

```diff
@@ -107,68 +100,98 @@ def loaded_grafana(docker_compose_runner, test_resources_dir):
         )
         yield docker_services
 
-    # The Grafana image can be large, so we remove it after the test.
     cleanup_image("grafana/grafana")
 
 
-@freeze_time(FROZEN_TIME)
-def test_grafana_dashboard(loaded_grafana, pytestconfig, tmp_path, test_resources_dir):
-    # Wait for Grafana to be up and running
-    url = "http://localhost:3000/api/health"
-    for i in range(30):
+def wait_for_grafana(url: str, max_attempts: int = 30, sleep_time: int = 5) -> bool:
+    """Helper function to wait for Grafana to start"""
+    for i in range(max_attempts):
```

Comment:
For my experience with other sources, this strategy on retries in the tests results flaky.
What worked better for me is to push it down to the docker compose, by defining proper service dependencies and more importantly the service health checks.

This does not need to necessarily be fixed now. Just an idea. Maybe a todo.